go.mod

```diff
@@ -13,7 +13,7 @@ require (
 	github.com/go-logr/logr v1.2.3
 	github.com/golang/mock v1.6.0
 	github.com/google/uuid v1.3.0
-	github.com/hashicorp/consul/api v1.15.2
+	github.com/hashicorp/consul/api v1.10.1-0.20221011200129-e2a6028e8e0f
```

Comment:
This PR depends on [fields](https://github.com/hashicorp/consul/blob/d17af23641d41fb7cfa56884711d071cf54b2a0d/api/peering.go#L75) that haven't yet been tagged/released. They will be included in an upcoming `api/v1.15.3` tag.